lib/bolt/outputter.rb

```diff
@@ -14,6 +14,18 @@ def self.for_format(format)
     def initialize(stream = $stdout)
       @stream = stream
     end
+
+    def unwrap_execution_result(result)
+      ok_hash = result.ok_nodes.iterator.map do |node, output|
```

Comment:
I originally started with checking whether result is an ExecutionResult. I'd still rather do that, but had trouble getting spec tests to correctly load the class.

Reply:
@thallgren calling `ok_nodes` or `error_nodes` - or using `TypeFactory.error.instance?` - all seem to cause an exception:
```
NoMethodError: undefined method `module_for_type' for nil:NilClass
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/p_object_type_extension.rb:27:in `create'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/puppet_object.rb:18:in `_pcore_type'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/type_calculator.rb:512:in `infer_Object'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:48:in `block in visit_this'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `each'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:42:in `visit_this'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/visitor.rb:82:in `visit_this_0'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/type_calculator.rb:252:in `infer'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/type_calculator.rb:271:in `infer_set'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/iterable.rb:77:in `block in on'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/iterable.rb:77:in `map'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/iterable.rb:77:in `on'
  /Users/michaelsmith/puppetlabs/bolt/vendored/puppet/lib/puppet/pops/types/execution_result.rb:72:in `iterator'
  /Users/michaelsmith/puppetlabs/bolt/lib/bolt/outputter.rb:21:in `unwrap_execution_result'
...
```

They go away if I force `TypeFactory.error.instance?` to get called during the plan run, such as adding a "noop" line like `$result.error_nodes` in the middle (and throwing away the result). Have any suggestions about how I should be checking for failure in ExecutionResult safely?

Reply:
Maybe it needs to happen within a `Puppet.override` block.